subprojects/build-cache/src/main/java/org/gradle/caching/local/internal/DirectoryBuildCacheServiceFactory.java

```diff
@@ -81,14 +84,15 @@ public BuildCacheService createBuildCacheService(DirectoryBuildCache configurati
         checkDirectory(target);
 
         int removeUnusedEntriesAfterDays = configuration.getRemoveUnusedEntriesAfterDays();
+        Supplier<Long> removeUnusedEntriesTimestamp = TimestampSuppliers.daysAgo(removeUnusedEntriesAfterDays);
```

Comment:
I'm totally ok with renaming this to something more meaningful, but I don't feel like `removeUnusedEntriesCreatedAfterTick` is a very good name.  For one, creation time is largely immaterial - the only thing that matters was when it was last accessed not necessarily when it was created.  Second, "after" is wrong - it's entries last accessed _before_ the timestamp that should be removed, not after.   And third, "tick" doesn't seem like a meaningful unit to me - when I hear tick, I think of something smaller than a second (i.e. a CPU tick).  

Some options:
- `removeUnusedEntriesOlderThan` is fine by me
- `removeEntriesLastAccessedBefore`?
- `unusedEntriesThreshold`?
- `keepEntriesLastAccessedAfter`?

Ultimately, I think the build cache configuration should match the configuration for the other GUH caches (i.e. timestamp property with convenience methods for specifying it in days) and potentially just implement some shared interface.  At that point, this would just look like the other caches.

Reply:
I changed this to `removeUnusedEntriesOlderThan` for now.  I'm open to changing it again, though.

Reply:
I think that name is already better.  I suppose tick isn't the right time, somehow I had it in my head that went with the epoch.  I think it would be useful to hint at the unit of measure in this name somehow, but it's already pretty long.